aspnetcore/mvc/views/tag-helpers/builtin/resources/CacheTagHelper.md

```diff
@@ -0,0 +1,286 @@
+```
+
+- - -
+
+### vary-by-route
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| String            | "Make"   	            |
+|                   | "Make,Model" |
+
+The attribute `vary-by-route` accepts a single header value or a comma-separated list of header values that trigger a cache refresh when they change. The example below looks at the values of `Make` and `Model`. When either or both change, the content of the Cache Tag Helper is rendered again, and the cache is reset.
+
+Usage Example:
+
+*Startup.cs* 
+
+```csharp
+routes.MapRoute(
+    name: "default",
+    template: "{controller=Home}/{action=Index}/{Make?}/{Model?}");
+```
+  
+*Index.cshtml*
+
+```html
+<Cache vary-by-route="Make,Model">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+- - -
+
+### vary-by-cookie
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| String            | ".AspNetCore.Identity.Application"   	            |
+|                   | ".AspNetCore.Identity.Application,HairColor" |
+
+The attribute `vary-by-cookie` accepts a single header value or a comma-separated list of header values that trigger a cache refresh when they change. The example below looks at the cookie associated with asp.net Identity. When a user is authenticated causing the request cookie to be set, The content of the Cache Tag Helper is rendered again, and the cache is reset.  
+
+Usage Example:
+
+```html
+<Cache vary-by-cookie=".AspNetCore.Identity.Application">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+````
+
+- - -
+
+### vary-by-user
+
+| Attribute Type 	| Example Values            	|
+|----------------	|----------------             	|
+| Boolean             | "true"   	            |
+|                     | "false" (default) |
+
+The attribute `vary-by-user` specifies whether or not the cache should reset when the logged-in user (or Context Principal) changes. The current user is also known as the  Request Context Principal and can be viewed in a Razor view by referencing `@User.Identity.Name`.
+
+The example below looks at the current logged in user. When a different user is found, The content of the Cache Tag Helper is rendered again, and the cache is reset.  
+
+Usage Example:
+
+```html
+<Cache vary-by-user="true">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+> [!NOTE]
+>  Using the attribute `vary-by-user` maintains the contents in cache through a log-in and log-out cycle.  When using `vary-by-cookie` which references the `.AspNetCore.Identity.Application` as shown above, a log-in and log-out action invalidates the cache for the same authenticated user because a new cookie value is generated. If no user is authenticated, that state is considered a valid cache state, which means that no logged-in user is one cache state, and the contents will be maintained for that condition as well.
```

Comment:
There are two spaces after the 1st sentence
There's a stray space before "Using ..."
Add comma before the "which" in "... which references the ..."